homeassistant/components/asuswrt/router.py

```diff
@@ -201,6 +254,52 @@ async def update_devices(self) -> None:
         if new_device:
             async_dispatcher_send(self.hass, self.signal_device_new)
 
+    async def update_sensors(self) -> None:
+        """Update AsusWrt sensors."""
+        self._sensors[SENSOR_CONNECTED_DEVICE].value = self._connected_devices
+
+        if self.has_sensor_type(AsusWrtSensorType.TypeBytes):
```

Comment:
If we use our coordinator update helper, it will handle updating only when there are connected callbacks (entities) and signaling. We can make one coordinator per api call.

https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

Reply:
May be I do not understand what you mean, but here is not the same? Api call are done only if entities are enabled. I always update the first sensors just because it use the information collected by the API in device tracking,

Reply:
The coordinator will only do automatic refresh if the entities are enabled. Is that what we want?

Reply:
Yes, this is what `has_sensor_type` is checking. May be I should rename it `has_sensor_type_enabled`?

Reply:
Right, so then we can use a coordinator instead that will give us this feature by default.

Reply:
I think that what you are asking for is already present in previous implementation. The only problem there is that we call the method used for `SENSOR_CONNECTED_DEVICE` 2 times but may be fixed with a different approach.

Reply:
Ok, thinking a little bit more on your idea, you are proposing to create multiple coordinator (1 for api) and use them to create the entities, that in this way will be managed separately and will call refresh method only when enabled. I will try to implement this on next w.e when I will have some more time available.